feature_engine/datetime/datetime.py

```diff
@@ -250,13 +275,36 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         # Check if input data contains same number of columns as dataframe used to fit.
         _check_input_matches_training_df(X, self.n_features_in_)
 
+        # special case index
+        if self.variables_ == ["index"]:
+            if self.missing_values == "raise":
+                if X.index.isnull().any():
+                    raise ValueError(
+                        "Index contain NaN. Check and "
+                        "remove those before using this transformer."
+                    )
+
+            for feat in self.features_to_extract_:
+                X[FEATURES_SUFFIXES[feat][1:]] = FEATURES_FUNCTIONS[feat](
+                    pd.Series(
+                        pd.to_datetime(
```

Comment:
converting something to datetime can take a lot of time if the dataset is big. Could we convert only if it is necessary instead of always?

Reply:
I notice that we do the same when the user passes variables. We convert to datetime all the time. Does pandas check that it is not datetime before doing the conversion? or converts everytime one says so, even though it could already be datetime?

Because in this case, it might be worth splitting the functionality. If not here, in a separate PR.

Reply:
Answering to my own question, to_datetime() does check if it is datetime already.

https://github.com/pandas-dev/pandas/blob/v1.4.1/pandas/core/tools/datetimes.py#L1027
https://github.com/pandas-dev/pandas/blob/06d230151e6f18fdb8139d09abf539867a8cd481/pandas/core/tools/datetimes.py#L289

So I don't think we need to change this. Apologies.

I also made some manual checks :p 

```
start = time.time()
data['Date_Time'] = pd.to_datetime(data['Date_Time'])
end = time.time()
print("The time of execution of above program is :", end-start)
The time of execution of above program is : 0.9556782245635986
```

And if i repeat that, after it was already converted to datetime, it takes a fraction:
`The time of execution of above program is : 0.013006925582885742`